src/network/core_utils.rs

```diff
@@ -192,6 +192,27 @@ impl CoreUtils {
         Ok(())
     }
 
+    #[tokio::main]
+    pub async fn turn_up_interface(ifname: &str) -> Result<(), Error> {
+        let (connection, handle, _) = match rtnetlink::new_connection() {
+            Ok((conn, handle, messages)) => (conn, handle, messages),
+            Err(err) => {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!("failed to connect: {}", err),
+                ))
+            }
+        };
+
+        tokio::spawn(connection);
```

Comment:
Not for this PR but we should stop doing it. We should only create one netlink connection and use this for all calls. Right now we have way to many unneeded clone syscalls.

Reply:
Sure we could pass it along the function. I'll refactor this later after making sure nothing breaks.